### PR TITLE
{2023.06}[gfbf/2023b] pystencils 1.3.4

### PR DESCRIPTION
**Pull request by xinan1911**

### CODE REVIEW

**Comment by xinan1911** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by xinan1911** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen4

**Comment by bedroge** [comment]
```
ERROR: One or more files not found: pystencils-1.3.4-gfbf-2023b.eb
```

This was probably added after the latest EB version was released, which means that you will have to use a ~`--from-pr`~ `--from-commit`.

File: easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.0-2023b.yml [hunk 1]
```diff
@@ -58,3 +58,4 @@ easyconfigs:
       options:
         from-pr: 20201
   - OSU-Micro-Benchmarks-7.2-gompi-2023b.eb
+  - pystencils-1.3.4-gfbf-2023b.eb
```
Comment (bedroge) [inline easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.0-2023b.yml:61]:
```suggestion
  - pystencils-1.3.4-gfbf-2023b.eb:
      options:
        # see https://github.com/easybuilders/easybuild-easyconfigs/pull/20889
        from-commit: c66c4788a17f7e4f55aa23f9fdb782aad97c9ce7
```

**Comment by xinan1911** [comment]
> ```
> ERROR: One or more files not found: pystencils-1.3.4-gfbf-2023b.eb
> ```
> 
> This was probably added after the latest EB version was released, which means that you will have to use a ~`--from-pr`~ `--from-commit`.

@bedroge "--from-commit" is added after EasyBuild 4.9.1. Should I also added Pystencils to easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.2-2023b.yml instead of easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.0-2023b.yml?

**Comment by bedroge** [comment]
> @bedroge "--from-commit" is added after EasyBuild 4.9.1. Does it mean that I need to add Pystencils to easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.2-2023b.yml instead of easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.0-2023b.yml?

Ah, yes, I overlooked that. In general it's always best to stick to the easystack of the latest EB release.

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by bedroge** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen4
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

**Review by bedroge — APPROVED** [review]

**Comment by bedroge** [comment]
For some reason the tarballs include an EB hooks file, even though it wasn't modified in this PR. We should check why it's included before we merge the staging PRs.

**Comment by bedroge** [comment]
> /project/def-users/SHARED/jobs/2024.07/pr_641/14321

I did a diff on the `eb_hooks.py` used for these builds and the one in the repo:
```
$ diff eb_hooks.py /cvmfs/software.eessi.io/versions/2023.06/init/easybuild/eb_hooks.py
87,88c87
<     cpu_target = get_eessi_envvar('EESSI_SOFTWARE_SUBDIR')
<     memory_hungry_build_a64fx = cpu_target == CPU_TARGET_A64FX and self.name in ['Qt5']
---
>     memory_hungry_build_a64fx = self.name in ['Qt5']
```

In https://github.com/EESSI/software-layer/pull/632 we made some additional changes to the hooks after the tarballs had been generated, and we didn't rebuild them, which explains why there is a diff and why the hooks file was included in the tarballs of this PR. So, we can safely proceed with the ingestion.

**Pull request merged**